vyper/ast/nodes.py

```diff
@@ -916,6 +916,25 @@ def s(self):
         return self.value
 
 
+class HexBytes(Constant):
+    __slots__ = ()
+    _translated_fields = {"s": "value"}
+
+    def __init__(self, parent: Optional["VyperNode"] = None, **kwargs: dict):
+        super().__init__(parent, **kwargs)
+        if isinstance(self.value, str):
+            self.value = bytes.fromhex(self.value)
+
+    def to_dict(self):
+        ast_dict = super().to_dict()
+        ast_dict["value"] = f"0x{self.value.hex()}"
+        return ast_dict
+
+    @property
+    def s(self):
```

Comment:
i wonder what would happen if we removed all of these :thinking: 

out of scope for this PR but we should check if that is still needed, since it was basically originally introduced for backwards compatibility